File: dart/test/freja_test.dart
import 'package:idkollen_client/idkollen_client.dart';
import 'package:test/test.dart';

import 'mock_client.dart';

void main() {
  late MockClient mock;
  late IdkollenClient client;

  setUp(() {
    mock = MockClient();
    client = IdkollenClientBuilder('cid', 'sec').baseUrl('https://x.test').httpClient(mock).build();
  });

  test('auth pending', () async {
    mock.enqueue(200, {
      'status': 'PENDING',
      'id': 'f1',
      'autoStartToken': 'tok',
      'qrData': 'qr',
    });
    final result = await client.freja.auth(const FrejaAuthRequest());
    expect(result, isA<FrejaPending>());
    expect((result as FrejaPending).qrData, 'qr');
  });

  test('auth completed', () async {
    mock.enqueue(200, {
      'status': 'COMPLETED',
      'id': 'f1',
      'ssn': '199001011234',
      'country': 'SE',
      'name': 'Test',
      'givenName': 'T',
      'surname': 'est',
    });
    final result = await client.freja.authStatus('f1');
    expect(result, isA<FrejaCompleted>());
    expect((result as FrejaCompleted).country, 'SE');
  });
}

File: ruby/test/freja_test.rb
# frozen_string_literal: true

require_relative 'test_helper'

class FrejaTest < Minitest::Test
  def setup
    @mock = MockAdapter.new
    @client = Idkollen::ClientBuilder.new('cid', 'sec').http_client(@mock.connection).build
  end

  def test_auth_pending
    @mock.enqueue(200, { 'status' => 'PENDING', 'id' => 'f1', 'autoStartToken' => 'tok', 'qrData' => 'qr' })
    result = @client.freja.auth(Idkollen::Freja::AuthRequest.new)
    assert_kind_of Idkollen::Freja::Pending, result
    assert_equal 'qr', result.qr_data
  end

  def test_auth_completed
    @mock.enqueue(200, {
      'status' => 'COMPLETED', 'id' => 'f2', 'ssn' => '19900101-1234',
      'country' => 'SE', 'name' => 'Anna Svensson',
      'givenName' => 'Anna', 'surname' => 'Svensson',
    })
    result = @client.freja.auth_status('f2')
    assert_kind_of Idkollen::Freja::Completed, result
    assert_equal 'SE', result.country
  end
end
